packages/cli/src/Generate.ts

```diff
@@ -309,6 +315,8 @@ More information: https://pris.ly/d/client`
 ${boxedTryAccelerateMessage}
 ${getHardcodedUrlWarning(config)}${breakingChangesStr}${versionsWarning}`
         }
+        const hideHintMode = args['--no-hints'] || false
```

Comment:
Right now, the code still goes through all of the trouble of creating hint message just to throw it away at the last moment if `--no-hint` is set. Can we instead refactor the code so that those code paths would be completely skipped in `--no-hints` case?

Reply:
I added another level of indentations and a large else block that accomplishes this 😆